Makefile

```diff
@@ -2,15 +2,21 @@ help: ## make [option]
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
 build-image: ## Build docker CI image locally 
-	DOCKER_BUILDKIT=1 docker-compose -f docker-compose.yml build 
+	DOCKER_BUILDKIT=1 docker-compose -f docker-compose-local.yml build 
```

Comment:
Hmm, I kinda need to still be able to build the image locally. For example, when trying to integrate Valgrind, I modified the Dockerfile, and I did that locally. I would edit the Dockerfile, build the image locally, test and repeat (what I call the "inner loop"). I did not want nor need to go all the way and push the image to the registry, it would have taken me too long to push and pull at every edit.

Reply:
Oh, sorry, my bad. I thought you meant that I should pull the image from the registry instead of building it, I saw "pull" and I panicked  :)
Yes, you are right, the documentation is vague about the --pull option of the build command. I also did some quick tests, it seems that, when you use the --pull option, it will check all the base images, the "FROM"s (in our case, ubuntu:20.04) to see if what we have locally is the freshest and if it's not, it will fetch it; if not, it will use the ubuntu:20.04 base image that we already have. (It compares the local and remote sha256 hashes). Interesting fact: if you're offline and use the --pull option, it will complain, even if you already have the FROM on your comp. This --pull option seems to be like a guarantee that the image you want to build will be built on the freshest layers possible. Please tell me if you discover otherwise :)